File: PagingnationInRecyclerview/app/src/main/java/velmm/com/pagingnationinrecyclerview/api/MovieService.java
package velmm.com.pagingnationinrecyclerview.api;

import java.util.List;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Query;
import velmm.com.pagingnationinrecyclerview.model.Movie;

public interface MovieService {
    @GET("volley_array.json")
    Call<List<Movie>> getMovies();
}

File: MasterDetailsMotionLayoutAnimation/app/src/main/java/com/howtodoandroid/masterdetailsmotionlayoutanimation/MovieService.kt
package com.howtodoandroid.masterdetailsmotionlayoutanimation

import retrofit2.Call
import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory
import retrofit2.http.GET


interface MovieService {
    @get:GET("volley_array.json")
    val movies: Call<List<Movie>>
}

object ClientApi {
    private var movieService: MovieService? = null
    val client: MovieService?
        get() {
            if (movieService == null) {
                movieService = Retrofit.Builder()
                    .addConverterFactory(GsonConverterFactory.create())
                    .baseUrl("https://howtodoandroid.com/")
                    .build()
                    .create(MovieService::class.java)
            }
            return movieService
        }
}